### PR TITLE
Addressing issue 2127

### PR DESCRIPTION
**Pull request by dsgouda**
Relaxing validation rule to check for readonly properties 'name', 'id' and 'type' only. Addresses https://github.com/Azure/autorest/issues/2127

### CODE REVIEW

File: src/modeler/AutoRest.Swagger/Validation/ResourceModelValidation.cs [hunk 1]
```diff
@@ -44,33 +47,19 @@ public class ResourceModelValidation: TypedRule<Dictionary<string, Schema>>
         /// </summary>
         /// <param name="definitions">Operation Definition to validate</param>
         /// <returns>true if the resource model is valid.false otherwise.</returns>
-        public override bool IsValid(Dictionary<string, Schema> definitions)
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string, Schema> entity, RuleContext context)
         {
-            foreach(string key in definitions.Keys)
+            var resModelList = entity.Where(pair => pair.Key.ToLower().Equals("resource"));
```
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Validation/ResourceModelValidation.cs]: should we be applying this to the list of resources already computed for this spec, based on the definition we came up with? why would we apply it just to something named "resource"?

Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/ResourceModelValidation.cs]: I merely modified what was already implemented. Yes, we can consider our new definition of "resource" here but we will still need to traverse the model hierarchy to get the root model definition which should contain the "name", "id" and "type" properties.

File: src/core/AutoRest.Core/Properties/Resources.resx [hunk 2]
```diff
@@ -157,21 +157,6 @@ Modelers:
     TypeName: CompositeSwaggerModeler, AutoRest.CompositeSwagger
     </value>
   </data>
-  <data name="CodeGenerationFailed" xml:space="preserve">
```
Comment (olydis) [inline src/core/AutoRest.Core/Properties/Resources.resx]: NICE cleanup!!! 👍 👍 👍 👍 👍 👍 👍 👍 👍

File: src/modeler/AutoRest.Swagger/Validation/ResourceModelValidation.cs [hunk 3]
```diff
@@ -39,38 +42,59 @@ public class ResourceModelValidation: TypedRule<Dictionary<string, Schema>>
...
-                    if (resourceSchema == null || resourceSchema.Properties.Count == 0)
-                        return false;
-
-                    if (!this.validateSchemaProperty(resourceSchema, "id", true)   ||
-                        !this.validateSchemaProperty(resourceSchema, "name", true) ||
-                        !this.validateSchemaProperty(resourceSchema, "type", true) ||
-                        !this.validateSchemaProperty(resourceSchema, "location", false) ||
-                        !this.validateSchemaProperty(resourceSchema, "tags", false))
-                        return false;
+                    yield return new ValidationMessage(new FileObjectPath(context.File, context.Path.AppendProperty(rootResourceModel)), this, rootResourceModel);
```
Comment (olydis) [inline src/modeler/AutoRest.Swagger/Validation/ResourceModelValidation.cs]: 👍

File: src/modeler/AutoRest.Swagger/Validation/ResourceModelValidation.cs (hunk 1 above)
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Validation/ResourceModelValidation.cs]: @dsgouda do we need to enforce that this is the case? does it matter for this rule? or just the fact that the properties exist in the full set of properties of the model and that they are read-only is enough?

**Comment by dsgouda** [comment]
@azuresdkci Test this please

**Pull request merged**